gpflow/functions.py

```diff
@@ -285,11 +285,11 @@ def __call__(self, X: TensorType) -> tf.Tensor:
         )  # X = X[:,:-1]
 
         # split up X into chunks corresponding to the relevant likelihoods
-        x_list = tf.dynamic_partition(X, ind, len(self.meanfunctions))
```

Comment:
I mentioned this offline but we should probably add a test if this wasn't previously breaking the tests. Though I guess this notebook is now at least providing coverage...